src/Orleans/Streams/Core/StreamSubscriptionManagerAdmin.cs

```diff
@@ -8,22 +8,27 @@ namespace Orleans.Streams.Core
 {
     internal class StreamSubscriptionManagerAdmin : IStreamSubscriptionManagerAdmin
     {
-        private IStreamProviderManager providerManager;
-        public StreamSubscriptionManagerAdmin(IStreamProviderManager providerManager)
+        private IStreamSubscriptionManager explicitStreamSubscriptionManager;
+        public StreamSubscriptionManagerAdmin(IStreamProviderRuntime providerRuntime)
         {
-            this.providerManager = providerManager;
+            // using ExplicitGrainBasedAndImplicit pubsub here, so if StreamSubscriptionManager.Add(Remove)Subscription called on a implicit subscribed
+            // consumer grain, its subscription will be handled by ImplicitPubsub, and will not be messed into GrainBasedPubsub 
+            this.explicitStreamSubscriptionManager = new StreamSubscriptionManager(providerRuntime.PubSub(StreamPubSubType.ExplicitGrainBasedAndImplicit), 
```

Comment:
Suggest adding StreamSubscriptionManager to a dictionary by name, and looking up manager by name in GetStreamSubscriptionManager.